src/socket.cr

```diff
@@ -114,6 +114,11 @@ class Socket < IO
     yield result if result.is_a?(Exception)
   end
 
+  @[Deprecated("Use `#connect(addr, Time::Span?, &)` instead.")]
```

Comment:
Wouldn't this mean the warning will say to use the block version even if you're using the blockless version?

Reply:
I copied the other messages, but the deprecation message should be personalized I guess. Maybe?

> Use `Time::Span` for timeouts instead.

Reply:
Ideally each overload would have a warning specific to that overload. But also should make sure using the blockless overload doesn't trigger a warning for both since it delegates to the block version.